proto/asn1-pdu/mutated_x509_chain.proto

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+// This protobuf represents an X.509 certificate chain with a set of operations
+// to mutate the chain to create structural relationships between the
+// certificates.
+
+syntax = "proto2";
+
+import "x509_certificate.proto";
+
+package mutated_x509_chain;
+
+message MutatedX509Chain {
+  required X509Chain x509_chain = 1;
+  repeated Operation operations = 2;
+}
+
+// An X.509 Certificate Chain comprises a list of |certificates|.
+message X509Chain {
+  repeated x509_certificate.X509Certificate certificates = 1;
+}
```

Comment:
It's not clear you need a distinct message type for this type. It should just be fine to have "repeated x509_certificate.X509Certificate" within the main `MutatedX509Chain`

Reply:
Yes, I agree with you and that was the initial intention. The reason why I ended up putting it in its own message was because I wanted to have an `EncodeX509Chain` helper to DER encode each certificate, and if it wasn't it's own message, then the argument for the function would have been `google::protobuf::RepeatedPtrField` which seemed bulky.

I realize that's not a good justification though. So I accepted your change!